cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/SystemArtifactLoader.java

```diff
@@ -42,10 +48,12 @@ public Service get() {
           protected void doStart() {
             try {
               artifactRepository.addSystemArtifacts();
+              LOG.info("Successfully added system artifacts");
               // if there is no exception, all good, continue on
               notifyStarted();
             } catch (Exception e) {
               // transient error, fail it and retry
+              ERROR_LOG.warn("Exception while adding system artifacts", e);
```

Comment:
we can wrap with another retry strategy with a limit on number of retries. but right now i am not sure if we should give up after `n retries`